esp-hal/src/dma/pdma.rs

```diff
@@ -897,3 +906,39 @@ where
         );
     }
 }
+
+/// A marker for SPI-compatible type-erased DMA channels.
+pub struct AnyPdmaSpiChannel;
```

Comment:
super nit: I think the public one should be named `AnySpiDmaChannel` instead as it's more consistent with `Spi2DmaChannel`/`Spi3DmaChannel` name wise.